mne/decoding/receptive_field.py

```diff
@@ -0,0 +1,370 @@
+           Frontiers in Human Neuroscience 10, 604.
+           doi:10.3389/fnhum.2016.00604
+
+    .. [4] Holdgraf, C. R. et al. Rapid tuning shifts in human auditory cortex
+           enhance speech intelligibility. Nature Communications,
+           7, 13654 (2016). doi:10.1038/ncomms13654
+    """
+
+    def __init__(self, tmin, tmax, sfreq, feature_names=None,
+                 estimator=None, scoring='r2'):  # noqa: D102
+        self.feature_names = feature_names
+        self.sfreq = float(sfreq)
+        self.tmin = tmin
+        self.tmax = tmax
+        self.estimator = 0. if estimator is None else estimator
+        if scoring not in _SCORERS.keys():
+            raise ValueError('scoring must be one of %s, got'
+                             '%s ' % (_SCORERS.keys(), scoring))
+        self.scoring = scoring
+
+    def __repr__(self):  # noqa: D105
+        s = "tmin, tmax : (%.3f, %.3f), " % (self.tmin, self.tmax)
+        if isinstance(self.estimator, str):
+            estimator = self.estimator
+        else:
+            estimator = str(type(self.estimator))
+        s += "estimator : %s, " % str(estimator)
+        if hasattr(self, 'coef_'):
+            feats = self.feature_names
+            if len(feats) == 1:
+                s += "feature: %s, " % feats[0]
+            else:
+                s += "features : [%s, ..., %s], " % (feats[0], feats[-1])
+            s += "fit: True"
+        else:
+            s += "fit: False"
+        if hasattr(self, 'scores_'):
+            s += "scored"
+            if callable(self.scorer_):
+                s += " (%s)" % (self.scorer_.__name__)
+        return "<ReceptiveField  |  %s>" % s
+
+    def fit(self, X, y):
+        """Fit a receptive field model.
+
+        Parameters
+        ----------
+        X : array, shape (n_times[, n_epochs], n_features)
+            The input features for the model.
+        y : array, shape (n_times[, n_epochs], n_outputs)
+            The output features for the model.
+        """
+        from sklearn.linear_model import Ridge
+        from sklearn.base import is_regressor, clone
+        X, y = self._check_dimensions(X, y)
+
+        # Initialize delays
+        self.delays_ = _times_to_delays(self.tmin, self.tmax, self.sfreq)
+
+        # Define the slice that we should use in the middle
+        self.keep_samples_ = _delays_to_slice(self.delays_)
+
+        if isinstance(self.estimator, numbers.Real):
+            estimator = Ridge(alpha=self.estimator)
+        elif is_regressor(self.estimator):
+            estimator = clone(self.estimator)
+        else:
+            raise ValueError('`estimator` must be a float or an instance'
+                             ' of `BaseEstimator`,'
+                             ' got type %s.' % type(self.estimator))
+        self.estimator_ = estimator
+        _check_estimator(self.estimator_)
+
+        # Create input features
+        n_times, n_epochs, n_feats = X.shape
+        n_outputs = y.shape[-1]
```

Comment:
can you comment on the dimensionality of y, and why we can't have an explicit call (e.g. y.shape[1])